packages/typespec-go/test/azure-http-specs/azure/versioning/previewversiongroup/previewversion_client_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package previewversiongroup_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"previewversiongroup"
+	"previewversiongroup/fake"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	azfake "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPreviewVersionClient_GetWidget(t *testing.T) {
+	const fakeID = "widget-123"
+	fakeWidget := previewversiongroup.Widget{
+		ID:    to.Ptr(fakeID),
+		Name:  to.Ptr("Sample Widget"),
+		Color: to.Ptr("blue"),
+	}
+
+	server := fake.PreviewVersionServer{
```

Comment:
Why you use fake server to do the test? It could not be logged by spector dashboard.